AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -761,163 +785,212 @@ private void RenderObjectInfluenceRadius(DrawingContext drawingContext, List<Lay
+                            DrawTrueInfluenceRangePolygon(curLayoutObject, sgc, gridDictionary);
+                        }
+                        else
+                        {
+                            DrawInfluenceRangePolygon(curLayoutObject, sgc);
+                        }
+                    }
 
-                        ////////////////////////////////////////////////////////////////
-                        //Draw in width of object
-                        sgc.LineTo(_coordinateHelper.GridToScreen(new Point(topRightCorner.X, startPoint.Y), GridSize), stroked, smoothJoin);
+                    if (sg.CanFreeze)
+                    {
+                        sg.Freeze();
+                    }
+                    drawingContext.DrawGeometry(_lightBrush, _radiusPen, sg);
+                }
+            }
+        }
 
-                        //Draw quadrant 2
-                        //Get end value to draw from top-right of 2nd quadrant to bottom-right of 2nd quadrant
-                        startPoint = new Point(topRightCorner.X, topRightCorner.Y - influenceRange);
-                        var endPoint = new Point(topRightCorner.X + influenceRange, topRightCorner.Y);
+        private void DrawTrueInfluenceRangePolygon(LayoutObject curLayoutObject, StreamGeometryContext sgc, Dictionary<double, Dictionary<double, AnnoObject>> gridDictionary)
+        {
+            var stroked = true;
+            var smoothJoin = true;
 
-                        //Following the rules for quadrant 2 - go right and down
-                        var currentPoint = new Point(startPoint.X, startPoint.Y);
-                        while (endPoint != currentPoint)
-                        {
-                            currentPoint = new Point(currentPoint.X, currentPoint.Y + 1);
-                            sgc.LineTo(_coordinateHelper.GridToScreen(currentPoint, GridSize), stroked, smoothJoin);
-                            currentPoint = new Point(currentPoint.X + 1, currentPoint.Y);
-                            sgc.LineTo(_coordinateHelper.GridToScreen(currentPoint, GridSize), stroked, smoothJoin);
-                        }
+            var geometryFill = true;
+            var geometryStroke = true;
 
-                        ////////////////////////////////////////////////////////////////
-                        startPoint = endPoint;
-                        //Draw in height of object
-                        sgc.LineTo(_coordinateHelper.GridToScreen(new Point(startPoint.X, bottomRightCorner.Y), GridSize), stroked, smoothJoin);
+            var cellsInInfluenceRange = RoadSearchHelper.BreadthFirstSearch(
+                PlacedObjects.Select(o => o.WrappedAnnoObject),
+                Enumerable.Repeat(curLayoutObject.WrappedAnnoObject, 1),
+                o => o.InfluenceRange - 1,// reduce distance to get cells INSIDE influence range (not those which are touching the influence range)
+                gridDictionary: gridDictionary);
```

Comment:
nit: We have a named parameter when the other paramters are not named. Remove the named paramter.

Reply:
There is a optional parameter `inRangeAction` before `gridDictionary` so I need to specify parameter name (or swap them in definition).